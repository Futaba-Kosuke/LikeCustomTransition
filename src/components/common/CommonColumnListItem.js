/**
 * キャンペーンリスト表示用UIのベース部分を表示するコンポーネント
 */
import React from 'react';

//NativeBaseを使用したコンポーネントの呼び出し
import {
  ListItem,
  Left,
  Body,
  Right,
  Text,
  Button,
  Thumbnail
} from 'native-base';

//react-native-router-fluxのインポート宣言(Actionを使用)
import { Actions } from 'react-native-router-flux';

//一覧シンプルリスト表示用のベースコンポーネントの内容を定義する
const CommonColumnListItem = ({ archive }) => {

  //取得した引数（オブジェクト：{ archive }）を分割する
  const { id, title, catchcopy, image_url } = archive;

  //表示する要素を返す
  return (
    <ListItem thumbnail>
      <Left>
        <Thumbnail square size={80} source={image_url} />
      </Left>
      <Body>
        <Text>{title}</Text>
        <Text note>{catchcopy}</Text>
      </Body>
      <Right>
        <Button transparent onPress={ () => console.log("Work In Progress.") }>
          <Text>詳細</Text>
        </Button>
      </Right>
    </ListItem>
  );
};

//このコンポーネントをインポート可能にする
export default CommonColumnListItem;
