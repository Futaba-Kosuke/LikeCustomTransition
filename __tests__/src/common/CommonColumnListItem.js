import 'react-native';
import React from 'react';
import CommonColumnListItem from '../../../src/components/common/CommonColumnListItem';
import { getArchiveList } from '../../../src/stub/SampleDataStub';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('共通コンポーネント(コラムアーカイブ表示用部分)のスナップショットテスト', () => {
  const archive = getArchiveList();
  const tree = renderer.create(
    <CommonColumnListItem archive={archive[0]} onPress={ () => console.log("テスト") } />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
