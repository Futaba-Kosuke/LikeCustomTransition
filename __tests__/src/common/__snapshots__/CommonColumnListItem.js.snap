// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`共通コンポーネント(コラムアーカイブ表示用部分)のスナップショットテスト 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": null,
        "borderColor": "#c9c9c9",
        "flexDirection": "row",
        "marginLeft": 15,
        "paddingBottom": null,
        "paddingRight": null,
        "paddingTop": null,
      }
    }
    thumbnail={true}
    virtual={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "center",
          "flex": 0,
          "flexDirection": "row",
        }
      }
      virtual={undefined}
    >
      <Image
        size={80}
        source={1}
        square={true}
        style={
          Object {
            "borderRadius": 0,
            "height": 56,
            "width": 56,
          }
        }
        virtual={undefined}
      />
    </View>
    <View
      style={
        Object {
          "alignItems": null,
          "alignSelf": null,
          "borderBottomWidth": 0.5,
          "borderColor": "#c9c9c9",
          "flex": 1,
          "marginLeft": 15,
          "paddingBottom": 15,
          "paddingTop": 15,
        }
      }
      virtual={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
            "marginLeft": null,
            "marginRight": 10,
          }
        }
        virtual={undefined}
      >
        パスタのお店
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        note={true}
        style={
          Object {
            "color": "#808080",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "200",
            "marginLeft": null,
            "marginRight": 10,
          }
        }
        virtual={undefined}
      >
        美味しいパスタを楽しむことができます。
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "alignSelf": "stretch",
          "borderBottomWidth": 0.5,
          "borderColor": "#c9c9c9",
          "flex": 0,
          "justifyContent": "center",
          "paddingBottom": 15,
          "paddingLeft": null,
          "paddingRight": 15,
          "paddingTop": 15,
        }
      }
      virtual={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": null,
            "backgroundColor": "transparent",
            "borderColor": "#007aff",
            "borderRadius": 5,
            "borderWidth": null,
            "elevation": 0,
            "flexDirection": "row",
            "height": null,
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#007aff",
              "fontFamily": "System",
              "fontSize": 13,
              "lineHeight": 19,
              "marginLeft": 0,
              "marginRight": 0,
            }
          }
          virtual={undefined}
        >
          詳細
        </Text>
      </View>
    </View>
  </View>
</View>
`;
